### PR TITLE
[terra-menu] Dynamic resizing of sub-menu

### PR DESCRIPTION
**Pull request by lokesh-0813**
fixes : #952

### CODE REVIEW

**Comment by lokesh-0813** [comment]
Dynamic resizing can be seen in [this](https://terra-framework-deploye-pr-992.herokuapp.com/tests/terra-menu/menu/menu/sub-menu) test example.

Let me know your thoughts @jmsv6d @StephenEsser

File: packages/terra-menu/src/_MenuContent.jsx [hunk 1]
```diff
@@ -146,6 +143,13 @@ class MenuContent extends React.Component {
     }
   }
 
+  setPageHeight(node) {
```
Comment (jmsv6d) [inline packages/terra-menu/src/_MenuContent.jsx]: You shouldn't need to do this calculation. Popup should handle this without setting an explicit height on the content. You can change this https://github.com/cerner/terra-framework/pull/992/files#diff-8821ddb6189a1accf05f7aef26338373R344 to be 
`const contentHeight = this.props.isHeightBounded && '100%' ;
And update the content container's `fill` prop to be just `this.props.isHeightBounded`

**Comment by jmsv6d** [comment]
SubmenuResize

Is this acceptable behavior? @neilpfeiffer

File: packages/terra-menu/src/_MenuContent.jsx (hunk 1 above)
Comment (lokesh-0813) [reply packages/terra-menu/src/_MenuContent.jsx]: updated here b91205212bdd630b10eb4d5f5748cd63e33ac4a0

File: packages/terra-menu/src/terra-dev-site/test/menu/menu/SubMenu.test.jsx [hunk 2]
```diff
@@ -50,6 +50,18 @@ class SubMenu extends React.Component {
...
+              <Menu.Item text="Default 2.4" key="2.4" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 2.5" key="2.5" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 2.6" key="2.6" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 2.7" key="2.7" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 2.8" key="2.8" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 2.9" key="2.9" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 3.0" key="3.0" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 3.1" key="3.1" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 3.2" key="3.2" className="TestNestedMenuContent" />,
+              <Menu.Item text="Default 3.1" key="3.3" className="TestNestedMenuContent" />,
```
Comment (saket2403) [inline packages/terra-menu/src/terra-dev-site/test/menu/menu/SubMenu.test.jsx]: Nit: It should be `Default 3.3` instead of 3.1.

**Comment by nramamurth** [comment]
@lokesh-0813 Are there no _wdio_ screenshots that need to be updated with this change? If not, then I'd recommend adding one.

File: packages/terra-menu/src/terra-dev-site/test/menu/menu/SubMenu.test.jsx (hunk 2 above)
Comment (lokesh-0813) [reply packages/terra-menu/src/terra-dev-site/test/menu/menu/SubMenu.test.jsx]: Fixed here 9515f20110ed39ea657e076965e3ed0e74cfd39c

**Comment by lokesh-0813** [comment]
> @lokesh-0813 Are there no _wdio_ screenshots that need to be updated with this change? If not, then I'd recommend adding one.

Updated here 9515f20110ed39ea657e076965e3ed0e74cfd39c and here a87334b66361e2c03b09cbaff34af6611959f615

File: packages/terra-menu/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Changed
+* Removed fixedHeight logic from Menu.
```
Comment (emilyrohrbough) [inline packages/terra-menu/CHANGELOG.md]: Can we update this descriptions o it's meaningful to users? Maybe something like we removed the fix height in sub-menus or we allow sub-menu's height to be determined by it's content?

**Comment by emilyrohrbough** [comment]
@lokesh-0813  I am noticing different behavior of bound menus when the sub-menu is clicked. It seems the sub-menu is not longer bound. [Here is a gif](https://share.getcloudapp.com/xQu0xone) of the behavior seen in the deployment vs the existing behaviors found on https://engineering.cerner.com/terra-framework/tests/terra-menu/menu/menu/bounded-menu.

**Comment by lokesh-0813** [comment]
> @lokesh-0813 I am noticing different behavior of bound menus when the sub-menu is clicked. It seems the sub-menu is not longer bound. [Here is a gif](https://share.getcloudapp.com/xQu0xone) of the behavior seen in the deployment vs the existing behaviors found on https://engineering.cerner.com/terra-framework/tests/terra-menu/menu/menu/bounded-menu.

It's expected, since sub-menu has lesser menu-items compared to main-menu menu-items. Now sub-menu determines its height based on the number of menu-items it possess.

I have tweaked the Bounded Menu example(removed main-menu items and added more sub-menu items) to give you an idea of what's going on ->

Existing behaviour(master Branch)
currenttweak

Deployment behaviour
deploymentTweak

**Comment by emilyrohrbough** [comment]
> @emilyrohrbough It's expected, since sub-menu has lesser menu-items compared to main-menu menu-items. Now sub-menu determines its height based on the number of menu-items it possess.

I thought a bounded menu's height was determined by i's parent container, not its contents. Is that not the case?

File: packages/terra-menu/src/_MenuContent.jsx [hunk 4]
```diff
@@ -350,7 +346,7 @@ class MenuContent extends React.Component {
         role="dialog"
         onKeyDown={this.onKeyDown}
       >
-        <ContentContainer header={header} fill={this.props.isHeightBounded || this.props.index > 0}>
+        <ContentContainer header={header} fill={this.props.isHeightBounded}>
```
Comment (ShettyAkarsh) [inline packages/terra-menu/src/_MenuContent.jsx]: why was the index prop used here before ?

Comment (lokesh-0813) [reply packages/terra-menu/src/_MenuContent.jsx]: index > 0 would imply that it is sub-menu that is being rendered.

**Comment by lokesh-0813** [comment]
> > @emilyrohrbough It's expected, since sub-menu has lesser menu-items compared to main-menu menu-items. Now sub-menu determines its height based on the number of menu-items it possess.
> 
> I thought a bounded menu's height was determined by i's parent container, not its contents. Is that not the case?

cc : @neilpfeiffer

File: packages/terra-menu/src/_MenuContent.jsx (hunk 4 above)
Comment (ShettyAkarsh) [reply packages/terra-menu/src/_MenuContent.jsx]: so how we are handling this scenario now ? after removing this ?

**Comment by jmsv6d** [comment]
@emilyrohrbough A menu/popup is "bounded" depending on if the height or width is greater than that of the bounding container. With the menu having a dynamic height, the state of if it's bounded or not can change. This is how I would expect this to work for this situation, I don't think there is an efficient way to get around it, especially for the reverse situation that @lokesh-0813 showed. But it's one of the visual oddities that we should have ux verify if it's acceptable, to determine if this is the actual approach we want to take to fix the original issue of if the submenu is much larger than the main menu, the submenu can end up with a very small scrollable area.

**Comment by mjhenkes** [comment]
Closing until we identify the appropriate solution on the parent issue: #952

**Pull request closed without merging**